core/templates/dev/head/app.js

```diff
@@ -106,6 +106,12 @@ oppia.constant('DEFAULT_PROFILE_IMAGE_PATH', '/avatar/user_blue_72px.png');
 // page visited.
 oppia.constant('ENABLE_PROMO_BAR', true);
 
+oppia.constant('FEEDBACK_POPOVER_HTML', '<feedback-popup></feedback-popup>');
+oppia.constant('ANSWER_POPOVER_HTML',
+  '<div style="max-height: 300px; overflow-y: scroll;">' +
```

Comment:
Just as a note, putting HTML in JavaScript is actually a bad practice. Do you think it would be possible to use a templateUrl instead of a template string (and if so, can we do that instead)?

Reply:
We can't really use `templateUrl` in this case. One thing that comes to my mind is to have the HTML in JS as minimal possible (this is already done for the feedback popup `<feedback-popup></feedback-popup>`)

So would extracting answer popover into new directive and then just have `<answer-popup></answer-popup>` in JS be okay?

Reply:
Yep, I think that's fine if there's no alternative. For reference, could you please add the links/evidence explaining why we can't use `templateUrl`?

Reply:
For popovers, [this](http://angular-ui.github.io/bootstrap/versioned-docs/0.14.1/#/popover) component is used. I tried using `uib-popover-html` but it did not work as it does not support dynamic html as seen [here](https://github.com/angular-ui/bootstrap/issues/4923). To use `uib-popover-template`, I will have to include html inside script tags and put that script in html file. To avoid doing that I used '$templateCache'.